src/general/StringUtils.cs

```diff
@@ -41,4 +44,115 @@ public static string FormatNumber(this double number, bool withSuffix = true)
 
         return number.ToString("0.#", CultureInfo.CurrentCulture);
     }
+
+    /// <summary>
+    ///   Splits string into different chunks by whitespace.
+    /// </summary>
+    /// <remarks>
+    ///   Only handles a single whitespace ( ) and not tabs, multiple spaces, etc. For quoted
+    ///   substrings handling, this only considers double quotes (") and not apostrophes (').
+    /// </remarks>
+    /// <param name="input">String to split.</param>
+    /// <param name="ignoreWithinQuotes">Ignore whitespace within quotation marks.</param>
+    /// <returns>A list of the substrings, starting from the beginning of the input string.</returns>
+    public static List<string> SplitByWhiteSpace(string input, bool ignoreWithinQuotes)
+    {
+        var result = new List<string>();
+
+        var cutPosition = 0;
+        var insideQuote = false;
+
+        for (var i = 0; i < input.Length; ++i)
+        {
+            var character = input[i];
+
+            var isAnEscapeSequence = i > 0 && input[i - 1] == '\\';
+
+            if (character == '"' && !isAnEscapeSequence && !insideQuote && ignoreWithinQuotes &&
+                input.IndexOf("\"", i + 1, StringComparison.InvariantCulture) != -1)
```

Comment:
Why is this check here? Wouldn't it be safer to assume that if there is no ending `"` then rest of the string is within quotes?
Also this doesn't match the design, as we are already going to iterate over everything, I see this as pretty harmful to run a second loop to find the next `"` character.

Reply:
Yeah I think that makes sense, though this needs mentioning in the docs that it doesn't always consider enclosing quotes.